### PR TITLE
[Deepseek V3.2] Only skip Indexer logits computation when is_extend_without_speculative

### PR DESCRIPTION
**Pull request by hlu1**
## Motivation

Fixed a bug in https://github.com/sgl-project/sglang/pull/12788

## Modifications

We should only skip Indexer logits computation when forward_mode.is_extend_without_speculative() returns true, meaning that no cuda graph is involved. The `is_extend_without_speculative` was only added to the MHA path, but not the MLA path. So when MLA is used (on B200 for example) and mtp is on,, the bug is triggered.

## Accuracy Tests
on 8xB200
20-shot gsm8k with mtp on:
Before fix: Accuracy: 0.008
After fix: Accuracy: 0.947

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Review by Fridge003 — APPROVED** [review]

**Comment by Fridge003** [comment]
dpsk 3.2 test passed: https://github.com/sgl-project/sglang/actions/runs/19160112724/job/54777801302?pr=12816
